api/routes/affiliations/patch.js

```diff
@@ -45,18 +45,35 @@ module.exports = (app, { db = knex } = {}) => {
         .select('user_id')
         .where({ state_id: stateId, id })
         .first();
+        
+        if (userId === affiliationUserId) {
+          logger.error({
+            id: request.id,
+            message: `user ${request.user.id} is attempting to edit their own role`
+          });
+          response.status(403).end();
+          return;
+        }
+        
+        const { status, roleId } = request.body;
+        const audit = auditor(statusToAction(status), request);
+        
+        
+      // Lookup role name and set expiration date accordingly
```

Comment:
@thetif @knollfear I thought this would work but there's a problem with my approach here: We pass in `-1` when we are revoking or denying a role. So ill come up with a better way to handle that case. open to ideas

Also do we want to not set an expiration date when roles are denied/revoked?